cli/src/api_client/mod.rs

```diff
@@ -238,3 +238,61 @@ fn status_code_help<T: FnMut(&Response) -> String>(
 
     Err(anyhow::Error::msg(error_message_with_help))
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{env, time::Duration};
+
+    use axum::response::Response;
+    use tempfile::NamedTempFile;
+    use test_utils::{mock_logger, mock_sentry, mock_server::MockServerBuilder};
+    use tokio::time;
+
+    use super::ApiClient;
+
+    #[tokio::test(start_paused = true)]
+    async fn logs_and_reports_for_slow_api_calls() {
+        let mut mock_server_builder = MockServerBuilder::new();
+        let logs = mock_logger(None);
+        let (events, guard) = mock_sentry();
+
+        async fn slow_s3_upload_handler() -> Response<String> {
+            time::sleep(Duration::from_secs(11)).await;
+            Response::new(String::from("OK"))
+        }
+        mock_server_builder.set_s3_upload_handler(slow_s3_upload_handler);
```

Comment:
Now we have some really awesome ability to capture logs, Sentry events, and modify APIs for tests